### PR TITLE
Openshift controller: Handle all required secrets

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Makes the openshift controller manage all secrets used for Openshift

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/controller/openshift/openshift_controller.go [hunk 1]
```diff
@@ -397,6 +434,31 @@ func (r *Reconciler) deployments(ctx context.Context, osData *openshiftData) err
 	return nil
 }
 
+func (r *Reconciler) ensureNamespace(ctx context.Context, c *kubermaticv1.Cluster) error {
+	if c.Status.NamespaceName == "" {
+		if err := r.updateCluster(ctx, c.Name, func(c *kubermaticv1.Cluster) {
```
Comment (mrIncompetent) [inline api/pkg/controller/openshift/openshift_controller.go]: As i just realized the `updateCluster` function does not return the cluster object nor does it modify a passed cluster pointer,

doesn't this mean this function will try to create a namespace without name in the first execution?

File: api/pkg/resources/apiserver/tls-serving-certificate.go [hunk 2]
```diff
@@ -55,7 +63,7 @@ func TLSServingCertificateCreator(data resources.SecretDataProvider) resources.N
 					fmt.Sprintf("%s.%s.svc.cluster.local", resources.ApiserverInternalServiceName, data.Cluster().Status.NamespaceName),
 				},
 				IPs: []net.IP{
-					*externalIP,
+					net.ParseIP(externalIP),
```
Comment (mrIncompetent) [inline api/pkg/resources/apiserver/tls-serving-certificate.go]: Add a check to validate that `net.ParseIP` did not return `nil` - as it does when it fails parsing the ip

File: api/pkg/controller/openshift/openshift_controller.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/controller/openshift/openshift_controller.go]: Good point, that signature didn't make too much sense. Updated it to take a cluster pointer instead that will later on contain the updated cluster

File: api/pkg/resources/apiserver/tls-serving-certificate.go (hunk 2 above)
Comment (alvaroaleman) [reply api/pkg/resources/apiserver/tls-serving-certificate.go]: Fixed

**Pull request merged**